drivers/sensor/bq34110/bq34110.c

```diff
@@ -0,0 +1,371 @@
+/*
+ * Copyright (c) 2021 Linumiz
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_bq34110
+
+#include <init.h>
+#include <math.h>
+#include <string.h>
+#include <sys/util.h>
+#include <drivers/i2c.h>
+#include <sys/__assert.h>
+#include <sys/byteorder.h>
+#include <drivers/sensor.h>
+
+#include "bq34110.h"
+
+/* 5ms delay between each data flash write */
+#define BQ34110_DELAY	5
+/*
+ * @brief Read a register value
+ *
+ * Registers have an address and 16-bit value
+ *
+ * @param bq34110 Private data for the driver
+ * @param reg_addr Register address to read
+ * @param val Place to put the value on success
+ * @return 0 if success, or negative error code from I2C_API
+ */
+static int bq34110_cmd_reg_read(const struct device *dev,
+				uint8_t reg_addr, int16_t *val)
+{
+	int ret;
+	uint8_t i2c_data[2];
+	const struct bq34110_config *bq34110_cfg = dev->config;
+
+	ret = i2c_write_read(bq34110_cfg->i2c_dev, bq34110_cfg->i2c_addr,
+			     &reg_addr, sizeof(reg_addr), &i2c_data, 2);
+	if (ret < 0) {
+		LOG_ERR("Unable to read register");
+		return ret;
+	}
+
+	*val = ((i2c_data[1] << 8) | i2c_data[0]);
+
+	return 0;
+}
+
+/**
+ * @brief sensor value get
+ *
+ * @param dev BQ34110 device to access
+ * @param chan Channel number to read
+ * @param val Returns sensor value read on success
+ * @return 0 if successful
+ * @return -ENOTSUP for unsupported channels
+ */
+static int bq34110_channel_get(const struct device *dev,
+			       enum sensor_channel chan,
+			       struct sensor_value *val)
+{
+	float int_temp;
+	int32_t avg_power;
+	struct bq34110_data *bq34110 = dev->data;
+
+	switch (chan) {
+	case SENSOR_CHAN_GAUGE_TEMP:
+		int_temp = (bq34110->internal_temperature * 0.1);
+		int_temp = int_temp - 273.15;
```

Comment:
please document where constants like this come from

Reply:
This still needs to be done. A code comment with a datasheet reference would be sufficient